content/languages/javascript/authoring.md

```diff
@@ -162,11 +162,21 @@ The Codewars runner provides a set of preinstalled packages, which are available
 
 ### Reference solution
 
-_TODO_
+If the test suite happens to use a reference solution to calculate expected values (which [should be avoided](/authoring/guidelines/submission-tests/#reference-solution) when possible), or some kind of reference data like precalculated arrays, etc., it must not be possible for the user to redefine, overwrite or directly access its contents. To prevent this, it should be defined in a scope local to the testing function, an `it` or `describe` block.
+
+The reference solution or data ___must not___ be defined in the top-level scope of the test suite or in the [Preloaded code](/authoring/guidelines/preloaded/).
 
 ### Calling assertions
 
-_TODO_
+Chai provides a large set of useful [assertions](https://www.chaijs.com/api/assert/), but when used incorrectly, they can cause a series of problems:
+- Use of an assertion not suitable for the given case may lead to incorrect test results,
+- Incorrectly used assertions may produce confusing or unhelpful messages.
+
+To avoid the above problems, calls to assertion functions should respect the following rules:
+- The expected value should be calculated _before_ invoking an assertion. The `expected` parameter passed to the assertion should not be a function call expression, but a value calculated directly beforehand.
+- Appropriate assertion functions should be used for each given test. `assert.strictEqual` is not suitable in all situations. Use `assert.approximately` for floating-point comparisons, `assert.isOk` for tests on boolean values, and `assert.throws` to test error handling.
```

Comment:
Boolean things like `isOk` may be good for readability for a static test that is alone in its `it`. If we use custom messages or random tests, I think usual generic assertions should be more convenient for booleans.

Reply:
I think I've seen users being puzzled by messages like _"Expected `True` to be `False`"_ or something like that, and IIRC it was emitted by `assert_equals(true, false)` i am just not sure in what language. They usually went "WTF how can `true` be false?"_. So a custom message would have to be used anyway.

Generally, custom messages should be more common. Test feedback of "Expected 5 to equal 3" is generally considered kinda WTFy in itself :)

Reply:
Yes, and that's because messages are supposed to be read in appropriate context. If `it` says  "should return False for n = 0" or at least "should work for n = 0", then any message is understandable, be it "Expected True to be False" or just "Test failed". With proper messages, it's all about readability and then there's not one true appropriate assertion.
I don't remember if custom messages in JS overwrite default messages of they add to them, but then custom messages make `isOk` appropriate for any comparison that's done manually anyway...